creusot-contracts/src/logic/ra/auth.rs

```diff
@@ -0,0 +1,116 @@
+use crate::{
+    logic::ra::{
+        RA,
+        view::{View, ViewRel},
+    },
+    *,
+};
+
+struct AuthViewRel<T>(T);
+
+impl<T> ViewRel for AuthViewRel<T>
+where
+    T: RA,
+{
+    type Auth = T;
+    type Frag = T;
+
+    #[logic]
+    #[open]
+    fn rel(a: Self::Auth, f: Self::Frag) -> bool {
+        f.incl(a) && a.valid()
+    }
+
+    #[law]
+    #[open(self)]
+    #[requires(Self::rel(a, f1))]
+    #[requires(f2.incl(f1))]
+    #[ensures(Self::rel(a, f2))]
+    fn rel_mono(a: Self::Auth, f1: Self::Frag, f2: Self::Frag) {}
+}
+
+/// The 'authority' Resource Algebra.
+///
+/// This is a specialisation of [`View`], where:
+/// - both `Auth` and `Frag` are the same type
+/// - the relation between the two is that `Frag` must always be included in `Auth`
+pub struct Auth<T: RA>(View<AuthViewRel<T>>);
```

Comment:
Move this higher, because the comment is helpful to understand the definitions above.